_topics/en/op_checktemplateverify.md

```diff
@@ -131,30 +131,230 @@ see_also:
+  executed opcode that precedes it as part of its protection against
+  recursive covenants.  This is abnormal among opcodes
+  (`OP_CODESEPARATOR` in its interaction with `OP_CHECKSIG` and variants
+  being the only current opcode we're aware of that requires similar
+  tracking).  At least one reviewer argues that dealing with the
+  abnormal state requirements of CTV [adds][oconnor complexity] "much
+  more complexity to anyone trying to formally model Script semantics".
+
+- **Limited experimentation:** an [analysis][towns ctv usage] in 2022
+  found limited experimentation with CTV on [signet][topic signet],
+  although at least some additional experimentation has been performed
+  since then.
+
+- **Commonly requires exogenous fees:** CTV commitments to future
+  transactions may be created long before those transactions are
+  broadcast, making [fee estimation][topic fee estimation] challenging.
+  A common solution to this is to allow [CPFP fee bumping][topic cpfp]
+  and other types of [exogenous fee contribution][topic fee sourcing].  However,
+  at least one reviewer has [argued][todd exogenous] that widespread use
+  of exogenous fee sourcing puts Bitcoin's decentralization at risk due
+  to an incentive to [pay fees out of band][topic out-of-band fees].
+
+## Compositions of CTV and other consensus changes
+
+Many proposals have been made for using CTV in combination with other
+consensus changes.  Some of the proposals covered by Optech include:
+
+- **CTV and OP_VAULT:** the [BIP345][] `OP_VAULT` proposed soft fork is
+  designed (in its latest versions) to work directly with CTV for
+  committing to spending operations for creating dynamic [vaults][topic
+  vaults].
+
+- **OP_VAULT alternative using MATT and CTV:** the [MATT][topic acc]
+  proposed soft fork has been described as being able to almost emulate
+  `OP_VAULT` when combined with CTV and two other opcodes.  This proposal
+  appears mainly designed to showcase the flexibility of MATT rather
+  than suggest activating MATT and CTV instead of `OP_VAULT` and CTV.
+
+- **CTV and OP_CHECKSIGFROMSTACK:** the [OP_CHECKSIGFROMSTACK][topic
+  op_checksigfromstack] (CSFS) proposed soft fork allows verifying that
+  a signature on the stack commits to another item on the stack.  This
+  is in comparison to existing signature verification operations in
+  Bitcoin that verify that a signature commits to a large amount of data
+  and metadata for the executing transaction.  It's [proposed][news285
+  lnhance] that CTV and CSFS together can emulate the proposed
+  `SIGHASH_ANYPREVOUT` (APO) soft fork and so can enable versions of
+  [LN-Symmetry][topic eltoo] (eltoo) and [Ark][topic ark]-style
+  [joinpools][topic joinpools].
+
+## History
+
+CTV was originally proposed as `OP_CHECKOUTPUTSHASHVERIFY` (COSHV) and
+was briefly renamed to `OP_SECURETHEBAG` before becoming
+`OP_CHECKTEMPLATEVERIFY`.  The earliest public version of the proposal
```

Comment:
more precisely, each proposal has been "unchanged". they are 3 distinct opcode proposals, but COSHV and STB are withdrawn in favour of CTV. So CTV is the draft proposal that was formalized, implemented, and submitted to the BIP repo. the others were WIP.

This is a bit different than e.g. Taproot, where many changes were made to what taproot was over the years before final inclusion. CTV is unchanged.